api/tests/test_submit_job.py

```diff
@@ -70,5 +77,25 @@ def test_submit_job_with_empty_description(client):
     batch = [
         make_job(description='')
     ]
+    setup_requests_mock(batch)
     response = submit_batch(client, batch)
     assert response.status_code == status.HTTP_400_BAD_REQUEST
+
+
+@responses.activate
+def test_submit_job_granule_does_not_exist(client, table):
+    batch = [
+        make_job('S1B_IW_SLC__1SDV_20200604T082207_20200604T082234_021881_029874_5E38'),
+        make_job('S1A_IW_SLC__1SDV_20200610T173646_20200610T173704_032958_03D14C_5F2B')
+    ]
+    setup_requests_mock(batch)
+
+    login(client)
+    response = submit_batch(client, batch)
+    assert response.status_code == status.HTTP_200_OK
+
+    response = submit_batch(client, [make_job('S1A_IW_SLC__1SDV_20200610T173646_20200610T173704_032958_03D14C_5F2A')])
+    assert response.status_code == status.HTTP_400_BAD_REQUEST
+    assert response.json['title'] == 'Bad Request'
+    assert response.json['detail'] == 'Requested scenes could not be found: ' \
+                                      'S1A_IW_SLC__1SDV_20200610T173646_20200610T173704_032958_03D14C_5F2A'
```

Comment:
Think I'd combine these into a single request with some good granules and some bad granules to verify the error message only complains about the bad ones.  The `200` test doesn't add anything on its own over other test cases for submitting jobs.